File: GitHub/Models/PullRequestReview.cs
using Microsoft.Kiota.Abstractions.Serialization;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
namespace GitHubTodoDemo.GitHub.Models {
    /// <summary>
    /// Pull Request Reviews are reviews on pull requests.
    /// </summary>
    public class PullRequestReview : IAdditionalDataHolder, IParsable {
        /// <summary>The _links property</summary>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public PullRequestReview__links? _links { get; set; }
#nullable restore
#else
        public PullRequestReview__links _links { get; set; }
#endif
        /// <summary>Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.</summary>
        public IDictionary<string, object> AdditionalData { get; set; }
        /// <summary>How the author is associated with the repository.</summary>
        public GitHubTodoDemo.GitHub.Models.Author_association? Author_association { get; set; }
        /// <summary>The text of the review.</summary>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public string? Body { get; set; }
#nullable restore
#else
        public string Body { get; set; }
#endif
        /// <summary>The body_html property</summary>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public string? Body_html { get; set; }
#nullable restore
#else
        public string Body_html { get; set; }
#endif
        /// <summary>The body_text property</summary>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public string? Body_text { get; set; }
#nullable restore
#else
        public string Body_text { get; set; }
#endif
        /// <summary>A commit SHA for the review.</summary>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public string? Commit_id { get; set; }
#nullable restore
#else
        public string Commit_id { get; set; }
#endif
        /// <summary>The html_url property</summary>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public string? Html_url { get; set; }
#nullable restore
#else
        public string Html_url { get; set; }
#endif
        /// <summary>Unique identifier of the review</summary>
        public int? Id { get; set; }
        /// <summary>The node_id property</summary>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public string? Node_id { get; set; }
#nullable restore
#else
        public string Node_id { get; set; }
#endif
        /// <summary>The pull_request_url property</summary>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public string? Pull_request_url { get; set; }
#nullable restore
#else
        public string Pull_request_url { get; set; }
#endif
        /// <summary>The state property</summary>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public string? State { get; set; }
#nullable restore
#else
        public string State { get; set; }
#endif
        /// <summary>The submitted_at property</summary>
        public DateTimeOffset? Submitted_at { get; set; }
        /// <summary>The user property</summary>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public SimpleUser? User { get; set; }
#nullable restore
#else
        public SimpleUser User { get; set; }
#endif
        /// <summary>
        /// Instantiates a new PullRequestReview and sets the default values.
        /// </summary>
        public PullRequestReview() {
            AdditionalData = new Dictionary<string, object>();
        }
        /// <summary>
        /// Creates a new instance of the appropriate class based on discriminator value
        /// </summary>
        /// <param name="parseNode">The parse node to use to read the discriminator value and create the object</param>
        public static PullRequestReview CreateFromDiscriminatorValue(IParseNode parseNode) {
            _ = parseNode ?? throw new ArgumentNullException(nameof(parseNode));
            return new PullRequestReview();
        }
        /// <summary>
        /// The deserialization information for the current model
        /// </summary>
        public IDictionary<string, Action<IParseNode>> GetFieldDeserializers() {
            return new Dictionary<string, Action<IParseNode>> {
                {"_links", n => { _links = n.GetObjectValue<PullRequestReview__links>(PullRequestReview__links.CreateFromDiscriminatorValue); } },
                {"author_association", n => { Author_association = n.GetEnumValue<Author_association>(); } },
                {"body", n => { Body = n.GetStringValue(); } },
                {"body_html", n => { Body_html = n.GetStringValue(); } },
                {"body_text", n => { Body_text = n.GetStringValue(); } },
                {"commit_id", n => { Commit_id = n.GetStringValue(); } },
                {"html_url", n => { Html_url = n.GetStringValue(); } },
                {"id", n => { Id = n.GetIntValue(); } },
                {"node_id", n => { Node_id = n.GetStringValue(); } },
                {"pull_request_url", n => { Pull_request_url = n.GetStringValue(); } },
                {"state", n => { State = n.GetStringValue(); } },
                {"submitted_at", n => { Submitted_at = n.GetDateTimeOffsetValue(); } },
                {"user", n => { User = n.GetObjectValue<SimpleUser>(SimpleUser.CreateFromDiscriminatorValue); } },
            };
        }
        /// <summary>
        /// Serializes information the current object
        /// </summary>
        /// <param name="writer">Serialization writer to use to serialize this model</param>
        public void Serialize(ISerializationWriter writer) {
            _ = writer ?? throw new ArgumentNullException(nameof(writer));
            writer.WriteObjectValue<PullRequestReview__links>("_links", _links);
            writer.WriteEnumValue<Author_association>("author_association", Author_association);
            writer.WriteStringValue("body", Body);
            writer.WriteStringValue("body_html", Body_html);
            writer.WriteStringValue("body_text", Body_text);
            writer.WriteStringValue("commit_id", Commit_id);
            writer.WriteStringValue("html_url", Html_url);
            writer.WriteIntValue("id", Id);
            writer.WriteStringValue("node_id", Node_id);
            writer.WriteStringValue("pull_request_url", Pull_request_url);
            writer.WriteStringValue("state", State);
            writer.WriteDateTimeOffsetValue("submitted_at", Submitted_at);
            writer.WriteObjectValue<SimpleUser>("user", User);
            writer.WriteAdditionalData(AdditionalData);
        }
    }
}

File: php/GitHub/Models/PullRequestReview.php
<?php

namespace Baywet\Githubtododemo\Github\Models;

use DateTime;
use Microsoft\Kiota\Abstractions\Serialization\AdditionalDataHolder;
use Microsoft\Kiota\Abstractions\Serialization\Parsable;
use Microsoft\Kiota\Abstractions\Serialization\ParseNode;
use Microsoft\Kiota\Abstractions\Serialization\SerializationWriter;

/**
 * Pull Request Reviews are reviews on pull requests.
*/
class PullRequestReview implements AdditionalDataHolder, Parsable 
{
    /**
     * @var PullRequestReview__links|null $_links The _links property
    */
    private ?PullRequestReview__links $_links = null;
    
    /**
     * @var array<string, mixed>|null $additionalData Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
    */
    private ?array $additionalData = null;
    
    /**
     * @var AuthorAssociation|null $author_association How the author is associated with the repository.
    */
    private ?AuthorAssociation $author_association = null;
    
    /**
     * @var string|null $body The text of the review.
    */
    private ?string $body = null;
    
    /**
     * @var string|null $body_html The body_html property
    */
    private ?string $body_html = null;
    
    /**
     * @var string|null $body_text The body_text property
    */
    private ?string $body_text = null;
    
    /**
     * @var string|null $commit_id A commit SHA for the review. If the commit object was garbage collected or forcibly deleted, then it no longer exists in Git and this value will be `null`.
    */
    private ?string $commit_id = null;
    
    /**
     * @var string|null $html_url The html_url property
    */
    private ?string $html_url = null;
    
    /**
     * @var int|null $id Unique identifier of the review
    */
    private ?int $id = null;
    
    /**
     * @var string|null $node_id The node_id property
    */
    private ?string $node_id = null;
    
    /**
     * @var string|null $pull_request_url The pull_request_url property
    */
    private ?string $pull_request_url = null;
    
    /**
     * @var string|null $state The state property
    */
    private ?string $state = null;
    
    /**
     * @var DateTime|null $submitted_at The submitted_at property
    */
    private ?DateTime $submitted_at = null;
    
    /**
     * @var NullableSimpleUser|null $user A GitHub user.
    */
    private ?NullableSimpleUser $user = null;
    
    /**
     * Instantiates a new pullRequestReview and sets the default values.
    */
    public function __construct() {
        $this->setAdditionalData([]);
    }

    /**
     * Creates a new instance of the appropriate class based on discriminator value
     * @param ParseNode $parseNode The parse node to use to read the discriminator value and create the object
     * @return PullRequestReview
    */
    public static function createFromDiscriminatorValue(ParseNode $parseNode): PullRequestReview {
        return new PullRequestReview();
    }

    /**
     * Gets the AdditionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     * @return array<string, mixed>|null
    */
    public function getAdditionalData(): ?array {
        return $this->additionalData;
    }

    /**
     * Gets the author_association property value. How the author is associated with the repository.
     * @return AuthorAssociation|null
    */
    public function getAuthorAssociation(): ?AuthorAssociation {
        return $this->author_association;
    }

    /**
     * Gets the body property value. The text of the review.
     * @return string|null
    */
    public function getBody(): ?string {
        return $this->body;
    }

    /**
     * Gets the body_html property value. The body_html property
     * @return string|null
    */
    public function getBodyHtml(): ?string {
        return $this->body_html;
    }

    /**
     * Gets the body_text property value. The body_text property
     * @return string|null
    */
    public function getBodyText(): ?string {
        return $this->body_text;
    }

    /**
     * Gets the commit_id property value. A commit SHA for the review. If the commit object was garbage collected or forcibly deleted, then it no longer exists in Git and this value will be `null`.
     * @return string|null
    */
    public function getCommitId(): ?string {
        return $this->commit_id;
    }

    /**
     * The deserialization information for the current model
     * @return array<string, callable(ParseNode): void>
    */
    public function getFieldDeserializers(): array {
        $o = $this;
        return  [
            '_links' => fn(ParseNode $n) => $o->setlinks($n->getObjectValue([PullRequestReview__links::class, 'createFromDiscriminatorValue'])),
            'author_association' => fn(ParseNode $n) => $o->setAuthorAssociation($n->getEnumValue(AuthorAssociation::class)),
            'body' => fn(ParseNode $n) => $o->setBody($n->getStringValue()),
            'body_html' => fn(ParseNode $n) => $o->setBodyHtml($n->getStringValue()),
            'body_text' => fn(ParseNode $n) => $o->setBodyText($n->getStringValue()),
            'commit_id' => fn(ParseNode $n) => $o->setCommitId($n->getStringValue()),
            'html_url' => fn(ParseNode $n) => $o->setHtmlUrl($n->getStringValue()),
            'id' => fn(ParseNode $n) => $o->setId($n->getIntegerValue()),
            'node_id' => fn(ParseNode $n) => $o->setNodeId($n->getStringValue()),
            'pull_request_url' => fn(ParseNode $n) => $o->setPullRequestUrl($n->getStringValue()),
            'state' => fn(ParseNode $n) => $o->setState($n->getStringValue()),
            'submitted_at' => fn(ParseNode $n) => $o->setSubmittedAt($n->getDateTimeValue()),
            'user' => fn(ParseNode $n) => $o->setUser($n->getObjectValue([NullableSimpleUser::class, 'createFromDiscriminatorValue'])),
        ];
    }

    /**
     * Gets the html_url property value. The html_url property
     * @return string|null
    */
    public function getHtmlUrl(): ?string {
        return $this->html_url;
    }

    /**
     * Gets the id property value. Unique identifier of the review
     * @return int|null
    */
    public function getId(): ?int {
        return $this->id;
    }

    /**
     * Gets the _links property value. The _links property
     * @return PullRequestReview__links|null
    */
    public function getlinks(): ?PullRequestReview__links {
        return $this->_links;
    }

    /**
     * Gets the node_id property value. The node_id property
     * @return string|null
    */
    public function getNodeId(): ?string {
        return $this->node_id;
    }

    /**
     * Gets the pull_request_url property value. The pull_request_url property
     * @return string|null
    */
    public function getPullRequestUrl(): ?string {
        return $this->pull_request_url;
    }

    /**
     * Gets the state property value. The state property
     * @return string|null
    */
    public function getState(): ?string {
        return $this->state;
    }

    /**
     * Gets the submitted_at property value. The submitted_at property
     * @return DateTime|null
    */
    public function getSubmittedAt(): ?DateTime {
        return $this->submitted_at;
    }

    /**
     * Gets the user property value. A GitHub user.
     * @return NullableSimpleUser|null
    */
    public function getUser(): ?NullableSimpleUser {
        return $this->user;
    }

    /**
     * Serializes information the current object
     * @param SerializationWriter $writer Serialization writer to use to serialize this model
    */
    public function serialize(SerializationWriter $writer): void {
        $writer->writeEnumValue('author_association', $this->getAuthorAssociation());
        $writer->writeStringValue('body', $this->getBody());
        $writer->writeStringValue('body_html', $this->getBodyHtml());
        $writer->writeStringValue('body_text', $this->getBodyText());
        $writer->writeStringValue('commit_id', $this->getCommitId());
        $writer->writeStringValue('html_url', $this->getHtmlUrl());
        $writer->writeIntegerValue('id', $this->getId());
        $writer->writeStringValue('node_id', $this->getNodeId());
        $writer->writeStringValue('pull_request_url', $this->getPullRequestUrl());
        $writer->writeStringValue('state', $this->getState());
        $writer->writeDateTimeValue('submitted_at', $this->getSubmittedAt());
        $writer->writeObjectValue('user', $this->getUser());
        $writer->writeObjectValue('_links', $this->getlinks());
        $writer->writeAdditionalData($this->getAdditionalData());
    }

    /**
     * Sets the AdditionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     * @param array<string,mixed> $value Value to set for the AdditionalData property.
    */
    public function setAdditionalData(?array $value): void {
        $this->additionalData = $value;
    }

    /**
     * Sets the author_association property value. How the author is associated with the repository.
     * @param AuthorAssociation|null $value Value to set for the author_association property.
    */
    public function setAuthorAssociation(?AuthorAssociation $value): void {
        $this->author_association = $value;
    }

    /**
     * Sets the body property value. The text of the review.
     * @param string|null $value Value to set for the body property.
    */
    public function setBody(?string $value): void {
        $this->body = $value;
    }

    /**
     * Sets the body_html property value. The body_html property
     * @param string|null $value Value to set for the body_html property.
    */
    public function setBodyHtml(?string $value): void {
        $this->body_html = $value;
    }

    /**
     * Sets the body_text property value. The body_text property
     * @param string|null $value Value to set for the body_text property.
    */
    public function setBodyText(?string $value): void {
        $this->body_text = $value;
    }

    /**
     * Sets the commit_id property value. A commit SHA for the review. If the commit object was garbage collected or forcibly deleted, then it no longer exists in Git and this value will be `null`.
     * @param string|null $value Value to set for the commit_id property.
    */
    public function setCommitId(?string $value): void {
        $this->commit_id = $value;
    }

    /**
     * Sets the html_url property value. The html_url property
     * @param string|null $value Value to set for the html_url property.
    */
    public function setHtmlUrl(?string $value): void {
        $this->html_url = $value;
    }

    /**
     * Sets the id property value. Unique identifier of the review
     * @param int|null $value Value to set for the id property.
    */
    public function setId(?int $value): void {
        $this->id = $value;
    }

    /**
     * Sets the _links property value. The _links property
     * @param PullRequestReview__links|null $value Value to set for the _links property.
    */
    public function setlinks(?PullRequestReview__links $value): void {
        $this->_links = $value;
    }

    /**
     * Sets the node_id property value. The node_id property
     * @param string|null $value Value to set for the node_id property.
    */
    public function setNodeId(?string $value): void {
        $this->node_id = $value;
    }

    /**
     * Sets the pull_request_url property value. The pull_request_url property
     * @param string|null $value Value to set for the pull_request_url property.
    */
    public function setPullRequestUrl(?string $value): void {
        $this->pull_request_url = $value;
    }

    /**
     * Sets the state property value. The state property
     * @param string|null $value Value to set for the state property.
    */
    public function setState(?string $value): void {
        $this->state = $value;
    }

    /**
     * Sets the submitted_at property value. The submitted_at property
     * @param DateTime|null $value Value to set for the submitted_at property.
    */
    public function setSubmittedAt(?DateTime $value): void {
        $this->submitted_at = $value;
    }

    /**
     * Sets the user property value. A GitHub user.
     * @param NullableSimpleUser|null $value Value to set for the user property.
    */
    public function setUser(?NullableSimpleUser $value): void {
        $this->user = $value;
    }

}
